### DISCUSSION TITLE
Offline search POI categories

### DISCUSSION THREAD

**Discussion by Ciutnik** [devemux86/cruiser#195] [Q&A] [answered]
As [suggested](https://github.com/devemux86/cruiser/discussions/189#discussioncomment-6046556) I would like to ask for the possibility to select more POIs visible on the map, similarly to other navigations, or rather trip planners. Currently, it is possible to show "Tourist attractions" layer, but each of us finds something different attractive. The more so that Cruiser is heading towards universal navigation to various types of tourism - from walking, cycling, motorcycling and perhaps even along the river by canoeing in the future (using the new BRouter features) 🙂

As I plan trips using a tablet, which is also a navigation device, at the planning stage (more than during navigation) I would like to be able to see objects such as museums, open-air museums, castles and palaces, also (or maybe mainly) those in ruins, former military facilities and monuments of technology that I love. My idea is to be able to distinguish the visibility of the POI set during planning/browsing and during navigation to maintain map readability. But while driving, I would definitely like to see roadside picnic areas, which I see a lot on OSM maps, and which I like to spontaneously use during trips.

I realize the complexity of my request, but it might be worth considering for the future?

**Comment by devemux86** [comment]
@Ciutnik thanks for the idea!

With offline POIs, such features are already planned!

Now we have free-form search for any text you can imagine.

We could add search for POI categories: fuel stations, speed cameras, museums, restaurants, ...
and show all results with different icons on the map.

Also add new overlays for these POI categories that are displayed as you move / zoom the map.
Press the POI on the map to reveal more details and use them in route planning and navigation.

The important thing is that everything is done **offline**.

It is one of the next big features that I am already working on. :slightly_smiling_face:

**Reply by DolomitesTom** [reply]
Aldi, Lidl, Norma, Rewe, Edeka or other super markets on the map to quickly buy some drinks before entering the hotel 🙂.

**Comment by devemux86** [comment]
**Offline** search of POI categories with all results on the map will be available in the next Cruiser.
You will even be able to change the color of the POI!

Now let's collect some important categories to include in the suggestions screen.
The first candidates as in [Nearby POI in navigation](https://github.com/devemux86/cruiser/discussions/9) can be:
- Fuel station
- Charging station
- Speed camera
- Parking
- ?

**Reply by MotoUKRider** [reply]
Some from [CoPilot in this thread](https://github.com/devemux86/cruiser/discussions/74#discussioncomment-5259763) which I posted a while back, if this helps.

**Reply by Ciutnik** [reply]
Often the first choice is also health care (pharmacies, hospitals, clinics), security (police, fire brigade) and... food and drink 🙂

**Comment by devemux86** [comment]
Preview of the new **offline** search with POI categories.

- The suggested search categories appear when the search field is empty.
(image 1)

- Click on a category to perform an **offline** search in all its sub-categories.

- Sub-categories have their own icons and show name, address, _opening hours_, direction, distance.
(image 2)

- Click on "Show on map" at the top to display all POI results on the map.

- To delete the displayed POI, long press on any POI and select "Delete".

- Free form search is also available when writing anything in search field.

Please let me know how you see the below structure and if more or different POI are needed.

---

The POI categories and the OpenStreetMap tags of their sub-categories currently are:
- **Hotel**: [tourism=hotel](https://wiki.openstreetmap.org/wiki/Tag:tourism%3Dhotel), [tourism=motel](https://wiki.openstreetmap.org/wiki/Tag:tourism%3Dmotel), [tourism=hostel](https://wiki.openstreetmap.org/wiki/Tag:tourism%3Dhostel), [tourism=camp_site](https://wiki.openstreetmap.org/wiki/Tag:tourism%3Dcamp_site)
- **Restaurant**: [amenity=restaurant](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Drestaurant), [amenity=fast_food](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dfast_food)
- **Cafe**: [amenity=cafe](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dcafe)
- **Bar**: [amenity=bar](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbar), [amenity=biergarten](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbiergarten), [amenity=pub](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dpub)
- **Supermarket**: [shop=convenience](https://wiki.openstreetmap.org/wiki/Tag:shop%3Dconvenience), [shop=supermarket](https://wiki.openstreetmap.org/wiki/Tag:shop%3Dsupermarket), [shop=bakery](https://wiki.openstreetmap.org/wiki/Tag:shop%3Dbakery)
- **Fuel station**: [amenity=fuel](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dfuel)
- **Charging station**: [amenity=charging_station](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dcharging_station)
- **Speed camera**: [highway=speed_camera](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dspeed_camera)
- **Parking**: [amenity=parking](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dparking)
- **Tourism**: [tourism=attraction](https://wiki.openstreetmap.org/wiki/Tag:tourism%3Dattraction), [tourism=museum](https://wiki.openstreetmap.org/wiki/Tag:tourism%3Dmuseum), [ tourism=viewpoint](https://wiki.openstreetmap.org/wiki/Tag:tourism%3Dviewpoint), [historic=castle](https://wiki.openstreetmap.org/wiki/Tag:historic%3Dcastle), [historic=fort](https://wiki.openstreetmap.org/wiki/Tag:historic%3Dfort), [historic=ruins](https://wiki.openstreetmap.org/wiki/Tag:historic%3Druins)
- **Picnic**: [tourism=picnic_site](https://wiki.openstreetmap.org/wiki/Tag:tourism%3Dpicnic_site), [leisure=picnic_table](https://wiki.openstreetmap.org/wiki/Tag:leisure%3Dpicnic_table)
- **Bank**: [amenity=bank](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dbank), [amenity=atm](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Datm)
- **Hospital**: [amenity=hospital](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dhospital), [amenity=clinic](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dclinic)
- **Entertainment**: [amenity=theatre](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dtheatre), [amenity=cinema](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dcinema), [amenity=library](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dlibrary)
- **Water**: [amenity=drinking_water](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Ddrinking_water)

<img src="https://github.com/user-attachments/assets/1bec61d7-6a2a-4a43-897e-57fe60d4b813" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/6fb54898-d176-4a6d-ba89-d54ab1e9f008" width="50%"/>

**Reply by DolomitesTom** [reply]
looks good, will this be available in next Cruiser version? Also concerning searching, but in favorites. At the moment Cruiser doesn't "recall" which category I have used on last app usage. I always have to select the category first. In order to avoid too much, could the favorites selection always show the search in the last category and not always show the category overview by default?

**Reply by devemux86** [reply]
> will this be available in next Cruiser version?

It will be available in the next Cruiser update. :slightly_smiling_face:

**Comment by devemux86** [comment, marked as answer]
Offline search of POI categories and display the results on the map is available in [Cruiser 3.0.25](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

For more details and instructions please see above:
- https://github.com/devemux86/cruiser/discussions/195#discussioncomment-6451058

**Reply by kds112** [reply]
Thanks for the new POI search feature! Is there any option/button to clear the POIs in the map? After a few searches the display is quite crowded with blue POI signs...

**Reply by devemux86** [reply]
> Is there any option/button to clear the POIs in the map?

Long press on any POI and select "Delete".
It clears all displayed POI markers.

**Comment by kds112** [comment]
The "display in the map" feature is very useful for the results of a categories search. But if I search e.g. for a frequent street name like "Hauptstraße" I will get several results, but I can't display all results in the map. Is it possible? Currently I have to check them one after another. 
And another request, is it possible to remember/offer the last search strings? I have to type "Hauptstraße" again and again in order to find the desired result.

**Reply by devemux86** [reply]
@kds112 thanks for the suggestions.

> if I search e.g. for a frequent street name like "Hauptstraße" I will get several results, but I can't display all results in the map. Is it possible?

"Show on map" all results is currently available for the search with POI categories.
I could also implement it for the free form search, if you find it useful.

> is it possible to remember/offer the last search strings?

Search history is already in my list. :slightly_smiling_face: 
I wanted to implement the POI categories first and then see it.

**Reply by kds112** [reply]
"Show on map" for free form search seems to be useful. If I look for "high street" in Greater London, I will get lots of results and can decide on the map which one is the right one... And by the way thanks a lot for all the new features in the latest update!!!

**Comment by Ciutnik** [comment]
I noticed that each search creates a new layer. When the search areas overlap, the results are duplicated. **It's not a bad thing**, but maybe there is a simple way to filter out the ones that have already been found earlier?

<img src="https://github.com/devemux86/cruiser/assets/122647782/fa3f0fa6-5ec7-47d0-8b67-b74feac23921" width="40%"/>

**Reply by devemux86** [reply]
We could clear the map before overlay new search results, so each time only the latest search is displayed.
Is this a possible solution?

Or as you mention we will need something more advanced, like filtering of duplicates overlays.

**Comment by devemux86** [comment]
About the POI icons, do you like them? Do they have good visibility?

Remember: you can change their color in "Settings | Search".

Or do you prefer something like the round / filled icons of the favorites?

**Comment by Ciutnik** [comment]
> We could clear the map before overlay new search results, so each time only the latest search is displayed.

Oh no, don't let it happen automatically. Duplicate points are less of a problem than their absence 🙂 You can leave it as it is, so as not to complicate things! Surely you have more important implementations lined up. 

> About the POI icons, do you like them?

I like them very much. They are distinctly different from Favorites and icons on the map/theme 👍 Thanks!

**Reply by DolomitesTom** [reply]
The new category search and icons are super

**Comment by devemux86** [comment]
> Or do you mean search and find all POI regardless their category?

Yes. That's what I meant because I know it from another app. The search circle is really small (maybe 50m), but in addition to POIs, it also shows addresses, found businesses, services, etc. It's not a key feature right now, but maybe it's an idea for the future.

_Originally posted by @Ciutnik in https://github.com/devemux86/cruiser/discussions/247#discussioncomment-6499052._

**Reply by devemux86** [reply]
I can add another category "Nearby points of interest" at end of the list,
that will find all poi regardless their category inside the screen bounds.

---

Note that the offline search uses map bounds and max number of results, for performance reasons.
e.g. Germany has millions of POI. Although I heavily optimized the offline search in the last Cruiser,
phones are not the huge servers of the online search engines!

So if you want more relevant results, please _zoom in_ more.
The max number of results is changed in "Settings | Search".

Info: Navigation's nearby POI have no limits, as it searches the entire route path for POI.

**Comment by Ciutnik** [comment]
Since one picture is worth 1000 words, this is how it looks in NaviExpert:

<img src="https://github.com/devemux86/cruiser/assets/122647782/7eaa613c-499a-4e2a-802b-3518e13485ad" width="24%"/> <img src="https://github.com/devemux86/cruiser/assets/122647782/ff4f5f6d-7cde-4eef-b6ba-b9c598e231c1" width="24%"/> <img src="https://github.com/devemux86/cruiser/assets/122647782/bebf38b0-c713-4501-9072-bf45ead83ed3" width="24%"/> <img src="https://github.com/devemux86/cruiser/assets/122647782/5a2a114d-902d-4b60-8140-6a0e28d270d3" width="24%"/>

**Reply by devemux86** [reply]
@Ciutnik thanks for the screenshots.
So we mean a similar search. :slightly_smiling_face:

**Comment by Ciutnik** [comment]
In theory, `historic=castle` should appear in the search for the **Tourism** category. But for some reason this [castle](https://www.openstreetmap.org/relation/3608812) does not appear. Because a polygon and not a point, or something like that?

**Reply by devemux86** [reply]
The POI databases in Cruiser contain both nodes and polygons (with their centroids).

However this seems to be a relation? Not a (closed) way, maybe this is the reason.
With relations everything becomes too complicated...

It would be better to have the POI in OpenStreetMap, as real _**Points**_ of interest. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
Yes, it looks like relation. But what's interesting, I can see it on the map (OpenAndroMaps and custom theme)...

<img src="https://github.com/devemux86/cruiser/assets/122647782/3bd9798b-7062-4a5d-905a-d20fce041b68" width="33%"/>

> It would be better to have the POI in OpenStreetMap, as real Points of interest.

👍 Unfortunately, we have what we have 🙂

Edit: My favorite map app also shows this castle:

<img src="https://github.com/devemux86/cruiser/assets/122647782/65e953b1-6705-410c-b3fb-01e7834aa9e0" width="33%"/>

**Reply by devemux86** [reply]
> But what's interesting, I can see it on the map

Mapsforge offline maps contain relations.

I can also add relations to offline POI in the future.

**Reply by Ciutnik** [reply]
> I can also add relations to offline POI in the future.

But only and only if it does not negatively affect anything - including your precious time! I looked at some other castles I know of in the area, and they are all found and shown 🙂

**Reply by devemux86** [reply]
@Ciutnik there are several BIG features planned!

I will try my best as always. :slightly_smiling_face:

**Reply by Ciutnik** [reply]
While I'm bothering you with new POIs, maybe you'd like to introduce a new POI category like... "Exploration"?

When I had more strength, I ran around ruins and abandoned historical buildings, hiding Geocaches. I loved this fun. Also by organizing motorcycle orienteering rallies, I also considered things like:

Ruins:
- historic=ruins
- building=ruins

Bunkier:
- military=bunker
- building=bunker
- historic=fort

Currently I have all this in another excellent app, but it would be great to have it in Cruiser too, to look at a place like this sometimes. I always look at them with tenderness and, of course, respect and responsibility! 😃

**Reply by devemux86** [reply]
The historic=castle is now in Tourism category. Perhaps we need another category for such "historic" tags.

Note: currently our [poi-mapping](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-poi-writer/src/main/config/poi-mapping.xml) that creates the POI does not contain building tags.

The historic=ruins, military=bunker, historic=fort already exist in the poi, we can offer them easily in Cruiser.

> Geocaches

With [c:geo](https://www.cgeo.org)?

**Reply by Ciutnik** [reply]
> The historic=ruins, military=bunker, historic=fort already exist in the poi, we can offer them easily in Cruiser.

Yes please! Maybe it's easiest to add to the existing "Tourism" category?

> With c:geo?

Unfortunately not. Those were the times when, after Psions, a few Palms, I was playing with the HP iPaq. I will never forget that microscopic touchpad, and a matchbox-sized GPS module that plugs into the slot. I also had an analog TV module! 😆 Opencaching (an open variant of Geocaching) worked so-so over WAP (Wireless Application Protocol), connected to Nokia phone via IR... Oh God, how it all went wrong! 🤣 To quote Zorba the Greek, "what a beautiful catastrophe" 😉

**Reply by devemux86** [reply]
If anyone uses c:geo for geocaching, one of the navigation apps that work with c:geo is also Cruiser:

<img src="https://github.com/devemux86/cruiser/assets/3484020/003c725d-c56d-4ddd-8008-059ecfad7ea6" width="50%"/>

**Reply by devemux86** [reply]
> Maybe it's easiest to add to the existing "Tourism" category?

They will be included in "Tourism" in the next Cruiser update.

**Reply by devemux86** [reply]
@Ciutnik 

About the icons:
- Bunker: how about the [home_filled icon](https://fonts.google.com/icons?selected=Material+Icons:home_filled:&icon.style=Filled) or is there anything better?
- Ruins: I have not found anything, currently use the [tourism icon](https://fonts.google.com/icons?selected=Material+Icons:account_balance:&icon.style=Filled)

**Reply by Ciutnik** [reply]
Your bunker proposal is so "concrete", but probably too "home". On some map I have bunkers marked as black rectangles. Aside from the color, something like that:

[Capture](https://fonts.google.com/icons?selected=Material%20Symbols%20Outlined%3Acapture%3AFILL%400%3Bwght%40400%3BGRAD%400%3Bopsz%4048)

But I found some military motives:

[Explosion](https://fonts.google.com/icons?selected=Material%20Symbols%20Outlined%3Aexplosion%3AFILL%400%3Bwght%40400%3BGRAD%400%3Bopsz%4048) which looks like a fort
[Destruction](https://fonts.google.com/icons?selected=Material%20Symbols%20Outlined%3Adestruction%3AFILL%400%3Bwght%40400%3BGRAD%400%3Bopsz%4048)
[Swords](https://fonts.google.com/icons?selected=Material%20Symbols%20Outlined%3Aswords%3AFILL%400%3Bwght%40400%3BGRAD%400%3Bopsz%4048)
[Military tech](https://fonts.google.com/icons?selected=Material%20Symbols%20Outlined%3Amilitary_tech%3AFILL%400%3Bwght%40400%3BGRAD%400%3Bopsz%4048)
[Fort](https://fonts.google.com/icons?selected=Material%20Symbols%20Outlined%3Afort%3AFILL%400%3Bwght%40400%3BGRAD%400%3Bopsz%4048) which looks like a perfect castle

In fact, there is a problem with "Ruins". Maybe this one?

[WB shade](https://fonts.google.com/icons?selected=Material%20Symbols%20Outlined%3Awb_shade%3AFILL%400%3Bwght%40400%3BGRAD%400%3Bopsz%4048)

Edit: I removed unnecessary pictures.

**Reply by devemux86** [reply]
The castle and fort icons are already used for their tags.

@Ciutnik please look in the Material icons (not symbols): https://fonts.google.com/icons?icon.set=Material+Icons

**Reply by Ciutnik** [reply]
So again, this time in the right set.

As a "Bunker" I suggest a rectangle (filled or outlined):

[Rectangle](https://fonts.google.com/icons?selected=Material%20Icons%3Arectangle%3A), [Crop 5 4](https://fonts.google.com/icons?selected=Material%20Icons%3Acrop_5_4%3A)
![Rectangle](https://github.com/devemux86/cruiser/assets/122647782/302e74e5-cec4-426c-b5ed-305ed2222644) 
![Crop_5_4](https://github.com/devemux86/cruiser/assets/122647782/7e04458b-8a02-46f3-9370-985b93cbab9e)

or [Pentagon](https://fonts.google.com/icons?selected=Material%20Icons%3Apentagon%3A) (any military connotations? 😉)
![Pentagon](https://github.com/devemux86/cruiser/assets/122647782/ed227c8a-dcfb-4968-aa23-b3acbf512865)

As in the previous failed proposal, maybe as "Ruins":

[WB shade](https://fonts.google.com/icons?selected=Material%20Icons%3Awb_shade%3A)
![WB_shade](https://github.com/devemux86/cruiser/assets/122647782/8d98b1a6-51a3-4260-9ae9-ead3550dc5c5)

maybe [Apartment](https://fonts.google.com/icons?selected=Material%20Icons%3Aapartment%3A) looks more appropriate
![Apartment](https://github.com/devemux86/cruiser/assets/122647782/9f2da4de-69b1-4214-a085-f407e88d01d5) 

or [Location city](https://fonts.google.com/icons?selected=Material%20Icons%3Alocation_city%3A)
![Location_city](https://github.com/devemux86/cruiser/assets/122647782/a5092866-01d1-4f71-83b6-5d70c7eff021)

**Reply by devemux86** [reply]
@Ciutnik thanks for the icons!

We could use the pentagon and wb_shade as more related icons.

**Reply by devemux86** [reply]
The extra POI in tourism are available in [Cruiser 3.0.27](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by 1maxi2** [comment]
Bei der offline Suche z.B. Lebensmittel wird mir die Unterkategorie nicht angezeigt. Auch " zeige auf Karte " nicht. Was mache ich falsch?

**Reply by devemux86** [reply]
@1maxi2 

You need to download / activate a POI file for your region.
Then, if there is nothing in the search area, the POI categories will appear as in the picture:

Sie müssen eine POI-Datei für Ihre Region herunterladen / aktivieren.
Wenn dann nichts im Suchbereich steht, erscheinen die POI-Kategorien wie auf dem Bild:

<img src="https://github.com/devemux86/cruiser/assets/3484020/a39b105c-1629-43ef-8a49-c368547d900d" width="50%"/>

**Reply by 1maxi2** [reply]
Der Fehler ist wieder mal vor dem Gerät gesessen.🫣 Ich hatte die Poi heruntergeladen aber nicht aktiviert ?!
Jetzt funktioniert es 👍
Sorry für die Belästigung .

**Reply by devemux86** [reply]
@1maxi2 

All downloads are activated automatically.
We can deactivate them later if we don't need to use them.

Alle Downloads werden automatisch aktiviert.
Wir können sie später deaktivieren, wenn wir sie nicht benötigen.

**Comment by devemux86** [comment]
> I can add another category "Nearby points of interest" at end of the list,
that will find all poi regardless their category inside the screen bounds.

**Offline** search of nearby points of interest (without category filter) will be available in the next Cruiser:

<img src="https://github.com/devemux86/cruiser/assets/3484020/4a027075-cb25-4975-8d52-a1d524e3c647" width="50%"/><img src="https://github.com/devemux86/cruiser/assets/3484020/cffa1075-9298-42dd-8d48-af699b887bdd" width="50%"/>

**Reply by Ciutnik** [reply]
Someone already asked for it. 
Would it be possible to show any uncategorized nearby search on the map? For example, the "Show on map" button would be useful here:

<img src="https://github.com/devemux86/cruiser/assets/122647782/9edc4ae2-bb4b-4647-ae83-0af0b9fcf1e6" width="50%"/>

**Reply by devemux86** [reply]
> Would it be possible to show any uncategorized nearby search on the map? For example, the "Show on map" button would be useful here

It is already planned. :slightly_smiling_face:

**Reply by devemux86** [reply]
> Would it be possible to show any uncategorized nearby search on the map? For example, the "Show on map" button would be useful here

"Show on map" for all search results will be available in the next Cruiser:

<img src="https://github.com/devemux86/cruiser/assets/3484020/ccf4adb5-a465-4c6e-b240-09ca97acf51e" width="50%"/>

**Comment by devemux86** [comment]
> I can add another category "Nearby points of interest" at end of the list,
that will find all poi regardless their category inside the screen bounds.

> Would it be possible to show any uncategorized nearby search on the map? For example, the "Show on map" button would be useful here

Both are available in [Cruiser 3.0.27](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by DolomitesTom** [comment]
Works great, the search results from different searches even stay on the map what is good. What if I want to delete the blue search points again manually? How do I do this?

**Reply by devemux86** [reply]
@DolomitesTom long press on any search result on the map and select "Delete".

**Reply by DolomitesTom** [reply]
thanks, this works

**Comment by Ciutnik** [comment]
I'm sorry to bother you with the little things, but if something can be improved, then maybe it should be... 
Well, i noticed that when "Nearby points of interest" is selected, the white labels of touched points that don't have a name, contain geographic coordinates. Maybe it would be better if it was a key value? So for example in this case (this is `amenity=fountain`) a white label could show `Fountain`.

<img src="https://github.com/devemux86/cruiser/assets/122647782/fe6485e2-5358-42f8-b730-d77819a48f20" width="50%"/>

_This is exactly [this](https://www.openstreetmap.org/node/4210576353) point._

**Reply by devemux86** [reply]
We should improve even the little things, so thanks for all suggestions! :+1: 

There are some technical things to consider:
- If we use the label "Fountain", should we translate in other languages or you only mean the English OSM value?
Translations are not easily possible, as they are not like the fixed categories, the labels can contain many values.
- Displaying `amenity=fountain` is a bit technical for most users, though they will at least see the category?
- Also what about the POI that have multiple key=value tags?
"amenity" is one of the possible OSM keys, not the only one.

**Reply by Ciutnik** [reply]
Of course, the ideal would be to translate (and even change some icons?), but there is probably no need! Automatic translation may not be correct 🙂 I found too `amenity=bicycle_parking`, `emergency=fire_hydrant`, `shop=greengrocer`, `historic=memorial`... There are a lot of categories, and significantly more values.

Technical `amenity=fountain` wouldn't be bad if it wasn't so long 😀

**Reply by devemux86** [reply]
Another idea could be to skip the unnamed POI when we search for "Nearby points of interest"?
Although I do not know if it is a good idea.

If you want to see the numerous tags included in POI files, see [here](https://github.com/mapsforge/mapsforge/blob/master/mapsforge-poi-writer/src/main/config/poi-mapping.xml). :slightly_smiling_face:

**Reply by Ciutnik** [reply]
> Another idea could be to skip the unnamed POI when we search for "Nearby points of interest"?

I think we'll lose too much... I guess I would vote to use the key value in the tag (and in the case of multiple tags - the first one in the list). It's better than geographical coordinates.
By the way, I noticed that the POI search is case sensitive(?). When we enter "Fountain" in the POI search engine, it will not find the "fountain" in tags. Will only find if the word "Fountain" is part of the name. But if we enter "fountain" - it will find it in the OSM tag. On the other hand, "ountain" will only find the "fountain" tag and not in the name 🙃

**Reply by devemux86** [reply]
> I noticed that the POI search is case sensitive(?).

Search is already case insensitive on names.
Cruiser also supports normalized search (without accents)!

With tags we need an improvement in the SQL, it will be available in next Cruiser.

**Reply by devemux86** [reply]
@Ciutnik many thanks for searching all these things, so we can improve them! :+1:

**Reply by devemux86** [reply]
> I guess I would vote to use the key value in the tag (and in the case of multiple tags - the first one in the list). It's better than geographical coordinates.

We can check for popular OSM keys, like amenity, shop, tourism, etc. or use the first one.
(we can use more OSM keys in this naming list as we discover them, to avoid first one)

Then we can have the following result with POI having a name or at least a tag reference:
(most of these POI have no name)

<img src="https://github.com/devemux86/cruiser/assets/3484020/f78616db-9d58-441e-8217-7463edac7236" width="50%"/>

**Reply by devemux86** [reply]
We can start with these OSM tag keys for the unnamed POI:
(if you want more keys, please let me know)

amenity, emergency, highway, historic, leisure, military, place, shop, sport, tourism

**Reply by Ciutnik** [reply]
> Then we can have the following result with POI having a name or at least a tag reference

Maybe the key name (like "emergency", "amenity", "tourism") don't we need to see? Wouldn't it be simpler "fire hydrant", "fountain", "camp site"? 
While it's not a problem on the list, the white label on the map would be too long - especially if we thought about some activation (manually or automatically on approach) of it while navigating 🙂

**Reply by devemux86** [reply]
@Ciutnik thanks good point! I will change it.

**Reply by Ciutnik** [reply]
I think "naming the unnamed" based on the value of the key, could also apply to the current **category** search. 

Because when you search e.g. for the  "Tourism" category, Viewpoints - that often don't have a name of their own - take the name "Tourism" (with a camera icon). Maybe "Viewpoint" would be better in this case? This is just an example. I know, there is a problem of translation, and additional work, and the effect is rather cosmetic... 🤔

**Reply by devemux86** [reply]
> I think "naming the unnamed" based on the value of the key, could also apply to the current category search.

It is used automatically in all search modes.

> Maybe "Viewpoint" would be better in this case?

Category names are often generic, so better we use the tags for more specific identification.

> I know, there is a problem of translation, and additional work

We have not translations here, as tag values are too many to handle.
Maybe in the future we can collect the main ones and translate them.

> and the effect is rather cosmetic...

@Ciutnik it is a nice suggestion, as it improves the user experience! :+1:

**Comment by devemux86** [comment]
> I noticed that the POI search is case sensitive(?).

> use the key value in the tag (and in the case of multiple tags - the first one in the list). It's better than geographical coordinates.

Case-insensitive search on POI tags and naming unnamed POI with tags are Implemented in [Cruiser 3.0.29](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by Ciutnik** [reply]
As always, thank you so much for another big update!!! ❤️ 
Unfortunately, it wouldn't be a good idea to use the first tag in the list as a label. This may not have a good effect. As you can see, different cases are possible... I'm sorry I suggested it myself.

<img src="https://github.com/devemux86/cruiser/assets/122647782/dd5a81a7-2b04-436c-be15-33becb785d11" width="33%"/>

The example above is [this](https://www.openstreetmap.org/node/7499562881) point.

**Reply by devemux86** [reply]
@Ciutnik thanks for the report and the tests!

When the first tag is used, there is already a check and if the value is yes / no, then we display the key + value.

The check is currently not present when the key is among the known list shown above: https://github.com/devemux86/cruiser/discussions/195#discussioncomment-6628296
I will add the same check there too to show the key + value if the value is yes / no.

Are there more values (besides yes / no) that we should display the key + value for better understanding?

**Reply by Ciutnik** [reply]
> can you post the OpenStreetMap link of this element to see its tags?

There is a [link](https://www.openstreetmap.org/node/7499562881) below the picture 🙂 At first I thought it only affected a few bunkers, but [this gas station](https://www.openstreetmap.org/node/1798258554) also has an unexpected label but matches `shop=no`

<img src="https://github.com/devemux86/cruiser/assets/122647782/53b2d328-791e-4ba4-a9fe-3c52948a601d" width="33%"/>

**Reply by devemux86** [reply]
@Ciutnik thanks for the examples.

This is easy, an improvement is needed in the tags parser. :slightly_smiling_face:

**Reply by devemux86** [reply]
Improved implementation for POI without name is available in [Cruiser 3.0.30](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Comment by devemux86** [comment]
Fuel stations currently are provided by the tag [amenity=fuel](https://wiki.openstreetmap.org/wiki/Tag:amenity%3Dfuel).

Do you also need the vending machines, like [vending=fuel](https://wiki.openstreetmap.org/wiki/Tag:vending%3Dfuel)?

(both in map and poi)

**Reply by DolomitesTom** [reply]
Sure, everything that prevents from running empty 😄
